### PR TITLE
test(partial-encryption): lock in the push/lock filename guard e2e and in docs (#467)

### LINKED ISSUE #467: Merged #443 follow-ups: partial-encryption still bypasses secret-lockout guard (#457) + init.md isidentifier docs stale (#460)
_🤖 Filed by the envdrift review-bot (via Jainal's account)._

Two review-bot findings from the #443 wave merged without being addressed. Both still reproduce on `main`.

---

## 1. Secret-lockout guard is backend-only — partial-encryption (`push`/`lock`) still bypasses it  ·  _medium · data-loss_

Originally raised on #457 (merged `6a60a63`). #457 added a filename-slug guard to **`DotenvxEncryptionBackend.encrypt`** (`src/envdrift/encryption/dotenvx.py`, the `[A-Za-z0-9._-]` check), and the merge also routed `DotenvxWrapper.encrypt` through `self._validate_filename`. **But `_validate_filename` only rejects the Windows `.env.local` pattern**, not space/non-ASCII slugs:

```python
# src/envdrift/integrations/dotenvx.py (main)
PROBLEMATIC_FILENAME_PATTERNS = [
    r"^\.env\.local$",          # <- the ONLY pattern
]
def _validate_filename(file_path):
    for pattern in PROBLEMATIC_FILENAME_PATTERNS:
        if re.match(pattern, file_path.name.lower(), re.IGNORECASE):
            raise DotenvxFilenameError(...)
```

`DotenvxWrapper.encrypt` has no `[A-Za-z0-9._-]` slug check of its own. The partial-encryption path uses the wrapper, not the guarded backend:

- `encrypt_secret_file` → `DotenvxWrapper().encrypt(secret_file)` (`src/envdrift/core/partial_encryption.py`)
- `_encrypt_secrets_only_file` (from `push_secrets_only`) → `dotenvx.encrypt(file)`, where `file` comes from `secrets_dir.glob(pattern)` — **arbitrary on-disk filenames**.

So a user with e.g. `my secret.env` or `café.env.secret` in their `secrets_dir` who runs `envdrift push` / `lock` (secrets-only) still hits the exact permanent-lockout this PR set out to fix: the value encrypts, dotenvx exits 0, the plaintext is destroyed, and the derived `DOTENV_PRIVATE_KEY_<bad slug>` is unusable so decrypt can never recover it. (The original finding empirically confirmed `roundtrip=0` for such names with dotenvx.) The existing regression tests cover the **backend** path (`test_encrypt_*` with `my secrets.env`/`café.env`) and café-in-*content*, but none drives the **partial-encryption** path with a space/unicode *filename*.

**Fix:** share the slug-validity guard so `DotenvxWrapper.encrypt` enforces it too (or have the partial-encryption callers route through the guarded backend), and add a regression test that drives `push_secrets_only`/`encrypt_secret_file` with a space/unicode-named secret file and asserts it refuses **with plaintext preserved**.

## 2. `docs/cli/init.md` `isidentifier()` claim contradicts the new leading-underscore aliasing  ·  _low · docs_

Originally raised on #460 (merged `ab127ad`). #460 made `init` sanitize a key that natively starts with `_` (e.g. `_PRIVATE` → `field__PRIVATE: str = Field(alias='_PRIVATE')`). But `docs/cli/init.md:178` still states:

> A key that passes `str.isidentifier()` (e.g. `CAFÉ`, `ΑΛΦΑ`) becomes a bare field with no alias.

A leading-underscore identifier passes `str.isidentifier()` yet is now aliased, so the claim is an over-generalization. Update the bullet (and add a leading-underscore example) so the docs match the shipped behavior (CLAUDE.md: keep `docs/` in sync with behavior changes).

### PR DESCRIPTION
**Pull request by jainal09**
## Summary

Follow-through on #467 (partial-encryption secret-lockout guard + stale docs). The two code/doc fixes
the issue asked for already merged — the wrapper-level filename guard covering the partial-encryption
push/lock seams in #468 (`cb68c97`), and the `docs/cli/init.md` `isidentifier()` correction in #469
(`212dbc4`). This PR closes what remained:

- **No end-to-end regression coverage existed.** The #468 regression tests are unit-level (in-process
  Python API; the guard fires pre-flight so they never touch dotenvx). Nothing drove the **real
  `envdrift push` CLI subprocess with the real dotenvx binary** through a guard-violating filename —
  the failure mode #467 describes. Two new integration tests (skip-gated on dotenvx/git, `tmp_path`
  isolated, explicit encodings, normalized output asserts + exit codes) now prove both modes refuse:
  - secrets-only mode with `my secret.env` (space → invalid dotenvx key slug)
  - combine mode with `café.env.secret` (non-ASCII → invalid key slug)

  Each asserts exit 1, a clean `Refusing to encrypt` error, plaintext preserved **byte-for-byte**, no
  combined file generated, and no `.env.keys` ever created (dotenvx never ran).
- **The refusal behavior was undocumented** (CLAUDE.md: keep docs in sync). `docs/cli/push.md` and
  `docs/cli/encrypt.md` now document the unsafe-filename refusal under *Safety guarantees*, pinned by
  a `test_docs_accuracy.py` test to the real `is_dotenvx_safe_filename` predicate so the docs cannot
  drift back.

## Fixes

Fixes #467

## Test evidence (fail-then-pass)

The defect no longer exists on this PR's base (it carries #468), so RED was proven on the pre-#468
commit (`cb68c97~1` — the last commit where #467 reproduced), GREEN on this branch:

**RED on `cb68c97~1`** — both new e2e tests fail; push reports success while destroying the secret:

```
assert result.returncode == 1
E AssertionError: ... ✓ Encrypted 1 file(s) in secrets ... [OK] Push complete! ...
E assert 0 == 1
FAILED ...::test_push_secrets_only_refuses_space_filename_real_cli
FAILED ...::test_push_combine_refuses_non_ascii_secret_filename_real_cli
```

(the file on disk was encrypted under an unusable `DOTENV_PRIVATE_KEY_<bad slug>` and the plaintext
was gone — the permanent lockout)

**GREEN on this branch**: `2 passed` (`pytest tests/integration/test_partial_encryption_e2e.py::TestUnsafeFilenameLockoutGuard`).

The docs-accuracy test was likewise proven RED against the unmodified docs
(`2 failed` — `permanently undecryptable` absent from both files), GREEN with the doc edits.

Gates: `ruff check` / `ruff format --check` / `pyrefly` / `bandit` clean; full non-integration suite
2536 passed; full `test_partial_encryption_e2e.py` module 25 passed; `make lint-docs` 0 errors.

## Notes

- **Stacked on #510 — merge that first.** Base is `fix/471-push-false-success`; this PR retargets to
  `main` automatically when #510 merges.
- **Overlap checked against #469** (merged): it fixed the `init.md` `isidentifier()` claim and added
  the leading-underscore/NFKC docs-accuracy tests — nothing is duplicated here; `init.md` needed no
  further change (verified against the shipped `_resolve_field_name` behavior).
- The diff is tests + docs only because the guard itself already merged in #468; this PR makes the
  #467 resolution regression-proof end to end per the repo's real-tests convention.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<a href="https://cubic.dev/pr/jainal09/envdrift/pull/525?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

<h3>Greptile Summary</h3>

This PR completes the #467 follow-through by adding two end-to-end integration tests that drive the real `envdrift push` CLI subprocess with the real dotenvx binary through guard-violating filenames (space and non-ASCII), and by documenting the refusal behaviour in `docs/cli/push.md` and `docs/cli/encrypt.md`, pinned by a new docs-accuracy test.

- Two `TestUnsafeFilenameLockoutGuard` integration tests assert exit 1, a \"Refusing to encrypt\" error, byte-for-byte plaintext preservation, and no `.env.keys` creation for both secrets-only (`my secret.env`) and combine (`café.env.secret`) push paths.
- `_write_combine_config` encoding was fixed to `utf-8` (addressing the previous review thread); `docs/cli/push.md` and `docs/cli/encrypt.md` each gain an unsafe-filename bullet pinned by the real `is_dotenvx_safe_filename` predicate.

<h3>Confidence Score: 5/5</h3>

Safe to merge — the changes are tests and docs only; no production code is modified.

The diff is entirely integration tests, a unit doc-accuracy test, and two markdown documentation updates. The guard logic itself lives in the already-merged #468. Previous review concerns (missing encoding in _write_combine_config, redundant byte assertions) are addressed. The one remaining inconsistency (_write_secrets_only_config lacking encoding=utf-8) does not affect any existing test.

tests/integration/test_partial_encryption_e2e.py — _write_secrets_only_config is now the only TOML writer in the module without an explicit encoding.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| tests/integration/test_partial_encryption_e2e.py | Adds TestUnsafeFilenameLockoutGuard with two e2e CLI tests; also fixes _write_combine_config to use encoding=utf-8. The analogous _write_secrets_only_config helper still lacks that encoding parameter. |
| tests/unit/test_docs_accuracy.py | Adds test_doc_documents_unsafe_filename_refusal parametrized over push.md/encrypt.md, pinning both docs to the real is_dotenvx_safe_filename predicate. |
| docs/cli/push.md | Adds unsafe-filename refusal bullet under Safety guarantees; wording matches the pinned doc-accuracy assertions. |
| docs/cli/encrypt.md | Adds the same unsafe-filename refusal bullet consistent with push.md. |

<h3>Sequence Diagram</h3>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
sequenceDiagram
    participant Test as Integration Test
    participant CLI as envdrift push (subprocess)
    participant Guard as is_dotenvx_safe_filename
    participant dotenvx as dotenvx binary

    Test->>CLI: run with space/non-ASCII secret filename
    CLI->>Guard: check filename
    Guard-->>CLI: False (invalid slug)
    CLI-->>Test: exit 1 + Refusing to encrypt
    Note over dotenvx: never invoked
    Test->>Test: assert plaintext unchanged
    Test->>Test: assert no .env.keys created
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
sequenceDiagram
    participant Test as Integration Test
    participant CLI as envdrift push (subprocess)
    participant Guard as is_dotenvx_safe_filename
    participant dotenvx as dotenvx binary

    Test->>CLI: run with space/non-ASCII secret filename
    CLI->>Guard: check filename
    Guard-->>CLI: False (invalid slug)
    CLI-->>Test: exit 1 + Refusing to encrypt
    Note over dotenvx: never invoked
    Test->>Test: assert plaintext unchanged
    Test->>Test: assert no .env.keys created
```

</a>

<sub>Reviews (12): Last reviewed commit: ["Merge branch &#39;main&#39; into fix/467-partial..."](https://github.com/jainal09/envdrift/commit/8122117fdd5c1e86817cbccb08c33a9655cc37f4) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=37244750)</sub>

### CODE REVIEW

File: tests/integration/test_partial_encryption_e2e.py [hunk 1]
```diff
@@ -86,14 +86,19 @@ def _write_combine_config(
     combined_file: str,
 ) -> None:
     """Write a combine-mode envdrift.toml for one environment."""
+    # encoding="utf-8" so a non-ASCII secret_file name (e.g. "café.env.secret")
```
Comment (jainal09) [inline tests/integration/test_partial_encryption_e2e.py:89]: Declining this CodeScene LOC advisory (non-required check) with rationale. The two new `TestUnsafeFilenameLockoutGuard` tests (+116 lines) tipped an already-near-limit e2e module over the 600-LOC default; they belong here because they reuse this module's real-dotenvx skip-gate and the `_run_envdrift`/`_write_combine_config`/`_out` helpers, so splitting them into a new file would duplicate ~45 lines of preamble and scatter cohesive partial-encryption e2e tests. This repo already carries much larger cohesive test modules by design (e.g. `tests/unit/test_cli.py` ~4537 LOC, `test_config_fail_loudly.py` ~461, `test_guard_exit_truthfulness.py` ~478), so the 600-line threshold is not enforced on test files. A dedicated module split, if desired, is better done as its own refactor than folded into this small tests+docs PR (#467).

[the remark this answers is not in the document]
Comment (jainal09) [reply tests/integration/test_partial_encryption_e2e.py:89]: Same non-required CodeScene LOC advisory as the sibling thread on this file — see the detailed rationale there. Short version: the two new TestUnsafeFilenameLockoutGuard tests tipped an already-near-limit e2e module over the 600-LOC default; they reuse this module's real-dotenvx skip-gate and _run_envdrift/_write_combine_config/_out helpers, so splitting them out would duplicate the preamble and scatter cohesive partial-encryption e2e tests. The repo already carries much larger cohesive test modules by design (test_cli.py ~4537 LOC), so the 600-line threshold isn't enforced on test files. Deferring the resolve decision to a human maintainer rather than self-resolving a declined advisory.

**Comment by jainal09** [comment]
@coderabbitai full review
